### ISSUE TITLE
Soviet 08a crashes on startup

### ISSUE THREAD

**Issue by RossBoylan** [OpenRA/OpenRA#21260]
## System Information

- Operating System:  Linux Debian 11.8
- OpenRA Version: ENGINE VERSION
- OpenRA Mod: Red Alert mod version release-20231010
- Source: Official download package

## Exception log

```
OpenRA engine version release-20231010
OpenRA Language: en
Red Alert mod version release-20231010
on map 40d54d93174bb69f39719165098b98c18a04a0eb (08a: Investigate Elba Island by Westwood Studios).
Date: 2023-12-17 02:46:03Z
Operating System: Linux (X64, Unix 5.10.0.26)
Runtime Version: .NET CLR 6.0.22
Installed Language: en (Installed) en (Current) en (Current UI)
Exception of type `System.ArgumentException`: Table already contains given key. (Parameter 'key')
   at Eluant.LuaTable.Add(LuaValue key, LuaValue value)
   at OpenRA.Scripting.ScriptContext..ctor(World world, WorldRenderer worldRenderer, IEnumerable`1 scripts) in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Scripting/ScriptContext.cs:line 222
   at OpenRA.Mods.Common.Scripting.LuaScript.OpenRA.Traits.IWorldLoaded.WorldLoaded(World world, WorldRenderer worldRenderer) in /home/runner/work/OpenRA/OpenRA/OpenRA.Mods.Common/Scripting/LuaScript.cs:line 45
   at OpenRA.World.LoadComplete(WorldRenderer wr) in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/World.cs:line 294
   at OpenRA.Game.StartGame(String mapUID, WorldType type) in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 202
   at OpenRA.Network.UnitOrders.ProcessOrder(OrderManager orderManager, World world, Int32 clientId, Order order) in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Network/UnitOrders.cs:line 177
   at OpenRA.Network.OrderManager.ReceiveImmediateOrders(Int32 clientId, OrderPacket orders) in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Network/OrderManager.cs:line 174
   at OpenRA.Network.NetworkConnection.OpenRA.Network.IConnection.Receive(OrderManager orderManager) in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Network/Connection.cs:line 350
   at OpenRA.Sync.<>c__DisplayClass13_0.<RunUnsynced>b__0() in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Sync.cs:line 167
   at OpenRA.Sync.RunUnsynced(World world, Action fn) in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Sync.cs:line 168
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager) in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 631
   at OpenRA.Game.LogicTick() in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 646
   at OpenRA.Game.Loop() in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 815
   at OpenRA.Game.Run() in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 868
   at OpenRA.Game.InitializeAndRun(String[] args) in /home/runner/work/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 306
   at OpenRA.Launcher.Program.Main(String[] args) in /home/runner/work/OpenRA/OpenRA/OpenRA.Launcher/Program.cs:line 32
```

## Additional information

I never see the game map; the crash is right away.

08b seems to work OK.

**Comment by JovialFeline** [comment]
Thank you for the report. This fortunately looks like the same error that https://github.com/OpenRA/OpenRA/pull/21251 has been merged to fix.

If it's not too much fuss and you'd like to be extra sure this is the same issue, you could load it in the map editor (Extras -> Map Editor -> Load Map -> 08a: Investigate Elba Island), rename the Radar Dome in that map to "Dome", save the map, and see if it works afterward.

**Comment by RossBoylan** [comment]
It certainly does look the same.  However, editing the map didn't seem to help.  I get the same trace as before, and the same crash.

Maybe it's not loading the modified map?  The save dialog says to save to user; maybe the original gets loaded?
After the first crash I went into the map editor again and confirmed that the name was now "Dome" for the radar dome.

Sorry I didn't see the original bug; I searched for 08a, but I guess that was only among open bugs.

Actually, the fix https://github.com/OpenRA/OpenRA/pull/21251/commits/acb0f7589460446fcfaf92d2e6eef71ccb8e6d1e looks as if it's changing the name of the key, not its value.  Maybe the former is not accessible through the map editor GUI.
